### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by zichen-3974**
Changes:
- Update UI component documentation.
- Update class diagram for UI.

See page [here](https://zichen-3974.github.io/tp/DeveloperGuide.html#ui-component)

### CODE REVIEW

**Review by seanfirefox — CHANGES_REQUESTED** [review]
LGTM! Just a minor change needed below.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -322,16 +322,23 @@ The **API** of this component is specified in [`Ui.java`](https://github.com/AY2
...
-The `UI` component uses the JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/AY2223S2-CS2103T-W14-2/tp/blob/master/src/main/java/seedu/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/AY2223S2-CS2103T-W14-2/tp/blob/master/src/main/resources/view/MainWindow.fxml)
 
 The `UI` component,
 
 * executes user commands using the `Logic` component.
 * listens for changes to `Model` data so that the UI can be updated with the modified data.
 * keeps a reference to the `Logic` component, because the `UI` relies on the `Logic` to execute commands.
-* depends on some classes in the `Model` component, as it displays `Person` object residing in the `Model`.
 
 {to be filled by Zichen}
```
Comment (russelltankaimin) [inline docs/DeveloperGuide.md:343]: Do you want to remove this as well?

**Review by russelltankaimin — CHANGES_REQUESTED** [review]
LGTM, just requesting a quick change.

**Review by seanfirefox — APPROVED** [review]
LGTM!

**Comment by russelltankaimin** [comment]
LGTM

**Review by russelltankaimin — APPROVED** [review]

**Pull request merged**